querybook/webapp/components/DataDocQueryCell/DataDocQueryCell.tsx

```diff
@@ -531,15 +531,42 @@ class DataDocQueryCellComponent extends React.PureComponent<IProps, IState> {
         reviewerIds: number[],
         externalRecipients: string[],
         notifierName: string,
-        justification: string,
-        query: string,
-        queryEngine: IQueryEngine
+        justification: string
     ) {
-        try {
-            console.error('TODO: Implement API call here');
-        } catch (error) {
-            console.error('Failed to request peer review:', error);
+        trackClick({
+            component: ComponentType.DATADOC_QUERY_CELL,
+            element: ElementType.PEER_REVIEW_QUERY_BUTTON,
+            aux: {
+                lintError: this.state.hasLintError,
+                sampleRate: this.sampleRate,
+            },
+        });
+
+        const transformedQuery = await this.getTransformedQuery();
+        if (!transformedQuery) {
+            return null;
         }
+
+        const executionMetadata =
+            this.sampleRate > 0 ? { sample_rate: this.sampleRate } : null;
+
+        const peerReviewParams = {
+            reviewer_ids: reviewerIds,
+            external_recipients: externalRecipients,
+            notifier_name: notifierName,
+            review_request_reason: justification,
+        };
+
+        const delayExecution = true;
+
+        await this.props.createQueryExecution(
+            transformedQuery,
+            this.engineId,
+            this.props.cellId,
+            executionMetadata,
+            peerReviewParams,
+            delayExecution
+        );
```

Comment:
this is quite similar to `onRunButtonClick`, can we refactor a bit to reuse and serve both?